ta/Makefile

```diff
@@ -4,25 +4,4 @@ CPPFLAGS += -DCFG_TEE_TA_LOG_LEVEL=$(CFG_TEE_TA_LOG_LEVEL)
 # The UUID for the Trusted Application
 BINARY=8aaaf200-2450-11e4-abe2-0002a5d5c51b
 
--include $(TA_DEV_KIT_DIR)/mk/ta_dev_kit.mk
```

Comment:
Ok. I'll move this to a separate commit.

The `all` and `clean` here overwrite those in `ta_dev_kit.mk`, causing a `No rule to make target '8aaaf200-2450-11e4-abe2-0002a5d5c51b.ta', needed by 'all'. ` error in the android build.

In trying to understand the error, it seems that `ta/Android.mk` calls `$(BUILD_OPTEE_MK)` which is `optee_os/mk/aosp_optee.mk` which calls `ta/Makefile` which calls `ta_dev_kit.mk` and friends and it seems like `8aaaf200-2450-11e4-abe2-0002a5d5c51b.ta` is not a valid target for `ta_dev_kit.mk` and friends.

If I change `all: $(BINARY).ta` to `all: $(link-out-dir)/$(BINARY).ta` here then there's no error in the android build since it's now a valid target for `ta_dev_kit.mk` and friends, namely `link.mk`.

BUT, why does `$(BINARY).ta` not cause a `No rule..` error when we use `build.git`? Any ideas?